pkg/core/data/ipc/transactions/payload_decoded.go

```diff
@@ -127,8 +131,35 @@ func (p *TransactionPayloadDecoded) Hash() ([]byte, error) {
 	return hashBytes, nil
 }
 
+func (p *TransactionPayloadDecoded) hashCoinbase() ([]byte, error) {
```

Comment:
As suggested, could we have unit test to ensure both hashCoinbase and Coinbase decoding does work?